### PR TITLE
feat: [CDS-56975]: Add quarterly recurrence support for deployment freeze

### PR DESCRIPTION
**Pull request by RishabhGupta34**
## Describe your changes
Add quarterly recurrence support for deployment freeze
Requires this pr to be merged first: https://github.com/harness/harness-go-sdk/pull/313
Tested in Prod and QA

## Comment Triggers

<details>
  <summary>PR Check triggers</summary>
  
- Build: `trigger build`
- Sub Category Field Check: `trigger subcategoryfieldcheck`

### CODE REVIEW

File: docs/data-sources/platform_manual_freeze.md [hunk 1]
```diff
@@ -81,5 +81,6 @@ Read-Only:
 Read-Only:
 
 - `until` (String)
+- `value` (Number)
```
Comment (rathodmeetsatish) [inline docs/data-sources/platform_manual_freeze.md:84]: description is not appearing for the fields

File: docs/resources/platform_manual_freeze.md [hunk 2]
```diff
@@ -107,6 +107,7 @@ Read-Only:
 Read-Only:
 
 - `until` (String)
+- `value` (Number)
```
Comment (rathodmeetsatish) [inline docs/resources/platform_manual_freeze.md:110]: description is not appearing for the fields

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]
Add a file in .changelog folder to describe the changes done in this pr

File: docs/data-sources/platform_manual_freeze.md (hunk 1 above)
Comment (RishabhGupta34) [reply docs/data-sources/platform_manual_freeze.md]: added

File: docs/data-sources/platform_manual_freeze.md [hunk 3]
```diff
@@ -80,6 +80,7 @@ Read-Only:
 
 Read-Only:
 
-- `until` (String)
+- `until` (String) Recurrence until timestamp
```
Comment (rathodmeetsatish) [inline docs/data-sources/platform_manual_freeze.md:83]: descriptions are still not reflect for all the fields

File: .changelog/528.txt [hunk 4]
```diff
@@ -0,0 +1,3 @@
+```release-note:enhancement
```
Comment (rathodmeetsatish) [inline .changelog/528.txt:1]: file name should be same as pr number

**Review by rathodmeetsatish — CHANGES_REQUESTED** [review]

File: docs/resources/platform_manual_freeze.md [hunk 5]
```diff
@@ -106,7 +106,8 @@ Read-Only:
 
 Read-Only:
 
-- `until` (String)
+- `until` (String) Recurrence until timestamp
```
[the remark this answers is not in the document]
Comment (RishabhGupta34) [reply docs/resources/platform_manual_freeze.md:109]: done

**Review by rathodmeetsatish — APPROVED** [review]

**Pull request merged**